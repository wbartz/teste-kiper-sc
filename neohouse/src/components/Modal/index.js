import PropTypes from 'prop-types';
import React from 'react';
import './index.scss';

const Modal = ({ title, className, children, show }) => {
  return (
    <>
      <div className={`custom-modal modal-overlay ${show ? 'active' : ''}`} />
      <div className={`modal custom-modal ${className}`}>
        <div className="modal-content">
          <h4>{title}</h4>
          <div className="content">{children}</div>
        </div>
      </div>
    </>
  );
};

Modal.propTypes = {
  title: PropTypes.string,
  children: PropTypes.node,
  show: PropTypes.bool,
  className: PropTypes.string,
};

Modal.defaultProps = {
  title: 'Ops!',
  children: null,
  show: true,
  className: '',
};

export default Modal;
