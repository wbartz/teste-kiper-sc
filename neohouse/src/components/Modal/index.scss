@import '../../styles/variables';

.custom-modal.modal-overlay {
  display: none;
}

.custom-modal.modal-overlay.active {
  z-index: 2;
  display: block;
  opacity: 0.5;
  background-color: $color-black;
  position: fixed;
}

.custom-modal.modal-overlay.active + .modal.custom-modal {
  position: absolute;
  z-index: 3;
  display: block;
  opacity: 1;
  top: 20%;
  transform: scaleX(1) scaleY(1);
}

.modal.custom-modal {
  width: 640px;
  padding: 20px;

  .modal-content {
    h4 {
      text-align: center;
      font-family: $font-family;
      font-size: 45px;
      font-weight: bold;
      color: $color-primary;
    }

    .content {
      font-family: $font-family-alternative;
      font-size: $h5-font-size;
      line-height: 1.5;
      letter-spacing: 0.15px;
      text-align: center;
      margin-top: 20px;
      color: $color-black;
    }
  }
}
