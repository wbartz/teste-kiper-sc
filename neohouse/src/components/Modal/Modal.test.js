import { shallow } from 'enzyme';
import React from 'react';
import Modal from '.';

describe('Modal', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Modal />);
  });

  it('render without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
