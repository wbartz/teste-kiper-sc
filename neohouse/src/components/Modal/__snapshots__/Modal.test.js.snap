// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal render without crashing 1`] = `ShallowWrapper {}`;
